### PR TITLE
python3 perf_compare.py diff_tests/perf_history/

### PR DESCRIPTION
**Pull request by colindelotavo**
Jobs have been updated to compare the latest performance logs before pushing to SVF-tools/Test-Suite.

### CODE REVIEW

File: .github/workflows/github-action.yml [hunk 1]
```diff
@@ -88,8 +88,15 @@ jobs:
       - name: ctest performance tests on cruxbc
         working-directory: ${{github.workspace}}/Release-build
         if: runner.os == 'Linux'
-        run: 
+        run: |
           ctest -R diff-perf-cruxbc -VV
+          # git pull
```
Comment (yuleisui) [inline .github/workflows/github-action.yml:93]: We need git pull first

Comment (colindelotavo) [reply .github/workflows/github-action.yml]: > We need git pull first

Uncommented in recent commit.

File: .github/workflows/github-action.yml [hunk 2]
```diff
@@ -88,8 +88,15 @@ jobs:
...
         if: runner.os == 'Linux'
-        run: 
+        run: |
           ctest -R diff-perf-cruxbc -VV
+          git pull
+          cp -r ${{github.workspace}}/Release-build/Testing/Temporary/LastTest.log $GITHUB_WORKSPACE/Test-Suite/diff_tests/perf_history/perf-$(date +'%Y-%m-%dT%H:%M:%S').txt
+          cp -r ${{github.workspace}}/Release-build/Testing/Temporary/LastTest.log $GITHUB_WORKSPACE/Test-Suite/diff_tests/perf-latest.txt
+          cd $GITHUB_WORKSPACE/Test-Suite/diff_tests
+          python3 -m pip install --upgrade pip setuptools wheel
+          pip3 install -r ./requirements.txt
```
Comment (yuleisui) [inline .github/workflows/github-action.yml:98]: can you wrap these in your python script?

Comment (yuleisui) [reply .github/workflows/github-action.yml:98]: Will these installations cause the CI to slow much

Comment (colindelotavo) [reply .github/workflows/github-action.yml]: If we really wanted to, we could wrap it in the python script, but this is not recommended, as described [here](https://pip.pypa.io/en/latest/user_guide/#using-pip-from-your-program). After adding `time` commands, I've found the installation time to only be a few seconds in total, so the runtime is negligible and shouldn't affect the CI runtime much.

File: .github/workflows/svf-lib_publish.yml [hunk 3]
```diff
@@ -94,8 +94,11 @@ jobs:
             git pull
             cp -r ${{github.workspace}}/Release-build/Testing/Temporary/LastTest.log $GITHUB_WORKSPACE/Test-Suite/diff_tests/perf_history/perf-$(date +'%Y-%m-%dT%H:%M:%S').txt
             cp -r ${{github.workspace}}/Release-build/Testing/Temporary/LastTest.log $GITHUB_WORKSPACE/Test-Suite/diff_tests/perf-latest.txt
-            cd $GITHUB_WORKSPACE/Test-Suite/diff_tests  
-            git add .     
+            cd $GITHUB_WORKSPACE/Test-Suite/diff_tests
+            python3 -m pip install --upgrade pip setuptools wheel
+            pip3 install -r ./requirements.txt
```
[the remark this answers is not in the document]
Comment (colindelotavo) [reply .github/workflows/svf-lib_publish.yml:99]: I've opened PR #913.

**Pull request closed without merging**